model/task/task.go

```diff
@@ -1220,8 +1220,14 @@ func (t *Task) MarkEnd(finishTime time.Time, detail *apimodels.TaskEndDetail) er
 
 func (t *Task) GetDisplayStatus() string {
 	if t.DisplayStatus != "" {
+		if t.Aborted {
```

Comment:
I don't think this branch is needed - if the display status is set then you should assume it's correct